analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Extensions/BinaryOperatorKindExtensions.cs

```diff
@@ -47,4 +49,56 @@ kind switch
             BinaryOperatorKind.LessThanOrEqual when left.Min > right.Max => BoolConstraint.False,
             _ => null
         };
+
+    public static NumberConstraint NumberConstraintFromRelationalOperator(this BinaryOperatorKind kind, NumberConstraint existingNumber, NumberConstraint comparedNumber) =>
+        kind switch
+        {
+            BinaryOperatorKind.Equals => NumberConstraint.From(
```

Comment:
I would probably add this on the `ArithmeticCalculator`, and just call it `FromRelationalOperator`, since it does mathematic calculations.

Reply:
I'm not opposed to that, but I think we are following Microsoft's recommendation to put extension methods in dedicated classes.

Reply:
With my suggestion, I would not keep it as an extension method, but as a normal instance one.
Both are fine to me, non-blocking.